src/python/pants/backend/python/tasks2/pytest_run.py

```diff
@@ -550,16 +557,137 @@ def _run_tests(self, targets):
       # A low-level test execution failure occurred before tests were run.
       raise TaskError()
 
-  def _do_run_tests(self, targets):
-    with self.invalidated(targets,
+  @staticmethod
+  def _vts_for_partition(invalidation_check):
+    return VersionedTargetSet.from_versioned_targets(invalidation_check.all_vts)
+
+  def check_artifact_cache_for(self, invalidation_check):
+    # We generate artifacts, namely junit.xml and coverage reports, that cover the full target set
+    # whether that is all targets in the context (`--fast`) or each target
+    # individually (`--no-fast`).
+    return [self._vts_for_partition(invalidation_check)]
+
+  @staticmethod
+  def _copy_dir(src, dst):
```

Comment:
Some explanation of how this method differs from alternatives would be good... not clear why it's here.